### DISCUSSION TITLE
Replace master/slave terminology

### DISCUSSION THREAD

**Discussion by Stezido** [openkfw/TruBudget#1043] [Ideas]
We should remove the master/slave terminology.

Alternatives could be following:

Master/Replica
Primary/Replica

I think **Replica** is a good alternative to slave because a node of a blockchain network is a Replica of other nodes in the network.
We could keep **Master** to not confuse Trubudget admins used to the master/slave terminology too much.
**Primary** is another alternative for master 

What's the Community thinking about it?

**Comment by jzakotnik** [comment]
Here we go, some ideas. Replica is nice, but it sounds like "fake art"?
- A and B
- Alpha and Beta (or Omikron :-)
- Chief and Participant
- Tree and Leaf

The advantage of Alpha and Beta is that it would enable also more complex hierarchies in the future (Gamma etc).

**Reply by dancingbenji** [reply]
- Base and Outpost
- Main and Dependent
- Origin and Manifestation
- Foundation and Characteristic

**Reply by Stezido** [reply]
I like the idea of the extendability of alpha beta omicron^^

**Comment by jzakotnik** [comment]
From others:
Root/Leave 
Main/Replicated
Planet/Satellite

**Comment by jzakotnik** [comment]
Conductor/follower
Primary/Secondary
Source/Sink

**Comment by jzakotnik** [comment]
Captain und Jack

**Comment by jzakotnik** [comment]
Head and Limb

**Comment by mathiashoeld** [comment]
i'm trying to capture my thought process, bare with me. 
first "level": talking about one node per organization
the first node has some kind of "unique power" since it creates the first blocks on the chain, right? so in that sense, i'd call it the "origin node". 
if another node tries to join the network, it has to be approved by a majority of all the nodes in the network. until then i'd call it a "requestor" or something similar. 
once the requestor is part of the network, it has the same rights as any other node, correct? if so, then i'd call every node on the network just "node", but then comes the second level. 

second "level": talking about multiple nodes per organization
each organization can have multiple nodes, correct? but adding one node to the network for one organization should not increase the organization's voting power on the net. therefore we should differentiate between "primary" and "secondary" nodes for an organization. 

in conclusion that would mean: 
- origin node: creator of the network -> should be renamed to primary node after another organization joins
- requestor: asks to join the network -> should be renamed as soon as request is granted
- primary node: regular node with full voting power
- secondary node: n-th node for each organization with no additional voting power

**Reply by Stezido** [reply]
Good thoughts!
Origin and requestor could be an alternative, but I wouldn't divide into primary and secondary nodes, because the voting power is not binded to a specific node just to the organization.
The origin/requestor division is more important because it changes it's behavior and configuration.
I would keep it short and simple 😚^^

**Reply by jzakotnik** [reply]
Thinking about this, I think we shouldn't connect the functionality to a particular naming. Sounds counter-intuitive, but if you have "lead + replica" or "origin + requester" or whatever, it gives room for misunderstanding of what it actually is. People won't read the description.  Therefore I think I favor something like alpha and beta, which you simply use and if you need more info, you look it up.

**Comment by Stezido** [comment]
Initiator/Follower
